### PR TITLE
[OAP-Cache][POAE7-345]Update oap cache  user doc and developer guide

### PR DESCRIPTION
**Pull request by jikunshang**
## What changes were proposed in this pull request?

update user_guide.md and developer_guide.md, move some build steps to developer guide.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by jikunshang** [comment]
@zhixingheyi-tian @yma11 Please take a review, thanks!

**Comment by yma11** [comment]
LGTM. Thanks!

**Comment by yma11** [comment]
LGTM. Thanks!

File: oap-cache/oap/docs/User-Guide.md [hunk 1]
```diff
@@ -228,44 +228,12 @@ The following are required to configure OAP to use DCPMM cache.
```
Comment (zhixingheyi-tian) [inline oap-cache/oap/docs/User-Guide.md:228]: Need "mkfs.ext4 /dev/pmem0 " before "mount -o dax "

Comment (jikunshang) [reply oap-cache/oap/docs/User-Guide.md:230]: Added.

**Comment by zhixingheyi-tian** [comment]
CC  @HongW2019

**Comment by HongW2019** [comment]
LGTM+1

**Comment by jikunshang** [comment]
> LGTM+1

Thanks! Will merge it.

**Pull request merged**